prov/verbs/src/verbs_mr.c

```diff
@@ -252,9 +252,10 @@ vrb_reg_dmabuf(struct vrb_domain *domain, const struct fi_mr_attr *attr,
 	 * to monitor or verify if the region is invalidated.
 	 */
 	ret = vrb_mr_nocache_reg(domain, NULL, attr->dmabuf->len, attr->access,
-				 attr->dmabuf->offset, attr->requested_key,
-				 flags, mr, attr->context, attr->iface,
-				 (uint64_t) attr->dmabuf->fd);
+				 (uint64_t) attr->dmabuf->base_addr +
+					 attr->dmabuf->offset,
```

Comment:
This won't work. In the end this will reach ibv_reg_dmabuf_mr() as offset. We need to pass both offset and virtual address down.

Reply:
you cannot pass iova as NULL when offset is not 0. They must have the same page offset

Reply:
Thanks. Updated PR.